### PR TITLE
[!feat] Refactor CryptoCompare

### PR DESCRIPTION
**Pull request by colmazia**
## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the GitHub PR explorer)

### CODE REVIEW

File: bothan-cryptocompare/src/api/builder.rs [hunk 1]
```diff
@@ -29,22 +29,33 @@ pub struct CryptoCompareRestAPIBuilder {
...
+        }
+    }
+
     /// Sets the URL for the API.
     /// The default URL is `DEFAULT_URL`.
-    pub fn set_url(&mut self, url: &str) -> &Self {
+    pub fn with_url(&mut self, url: &str) -> &Self {
         self.url = url.into();
         self
     }
```
Comment (warittornc) [inline bothan-cryptocompare/src/api/builder.rs:49]: please change to `mut self`

File: bothan-cryptocompare/src/api/builder.rs [hunk 2]
```diff
@@ -29,22 +29,33 @@ pub struct CryptoCompareRestAPIBuilder {
...
 
     /// Sets the API key for the API.
     /// The default is `None`.
-    pub fn set_api_key(&mut self, api_key: &str) -> &Self {
+    pub fn with_api_key(&mut self, api_key: &str) -> &Self {
```
Comment (warittornc) [inline bothan-cryptocompare/src/api/builder.rs:52]: ditto

File: bothan-cryptocompare/src/worker/asset_worker.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+            let result = timeout(
+                interval.period(),
+                update_asset_info(&worker.store, &worker.api, &ids),
+            )
+            .await;
+
+            if result.is_err() {
+                warn!("updating interval exceeded timeout")
+            }
```
Comment (warittornc) [inline bothan-cryptocompare/src/worker/asset_worker.rs:41]: pls make this more succinct if the result isn't going to be used

Comment (colmazia) [reply bothan-cryptocompare/src/worker/asset_worker.rs:41]: how na?? result is used to be check if it is err and this is code from coingecko source

File: bothan-cryptocompare/src/api/builder.rs (hunk 1 above)
Comment (colmazia) [reply bothan-cryptocompare/src/api/builder.rs]: fixed

File: bothan-cryptocompare/src/worker/asset_worker.rs (hunk 3 above)
Comment (warittornc) [reply bothan-cryptocompare/src/worker/asset_worker.rs:41]:
```
            let update_fn = update_asset_info(&worker.store, &worker.api, &ids);
            if timeout(interval.period(), update_fn).await.is_err() {
                warn!("updating interval exceeded timeout")
            }
            ```

File: bothan-cryptocompare/examples/cryptocompare_basic.rs [hunk 4]
```diff
@@ -1,20 +1,35 @@
...
+        .await
+        .unwrap();
+
+    worker.set_query_ids(vec!["BTC".to_string()]).await.unwrap();
+
+    sleep(Duration::from_secs(2)).await;
+
+    loop {
+        let data = worker.get_asset("BTC").await;
+        println!("{:?}", data);
```
Comment (RogerKSI) [inline bothan-cryptocompare/examples/cryptocompare_basic.rs:32]: Can add ETH? all examples shows query for two assets.

**Review by RogerKSI — CHANGES_REQUESTED** [review]

File: bothan-cryptocompare/examples/cryptocompare_basic.rs (hunk 4 above)
Comment (colmazia) [reply bothan-cryptocompare/examples/cryptocompare_basic.rs]: fixed

**Review by warittornc — APPROVED** [review]

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**